touchforms/formplayer/views.py

```diff
@@ -97,22 +112,26 @@ def enter_form(request, **kwargs):
         session_data['preloaders'] = preload_data
 
     if not xform:
-        xform = get_object_or_404(XForm, id=xform_id)
+        try:
+            xform = get_object_or_404(XForm, id=xform_id)
+        except ValueError:
```

Comment:
That's a surprising exception to come out of `get_object_or_404`. Seems like it maybe should be fixed deeper inside that function?